### ISSUE TITLE
[RFC]: don't run overlapping containers for non-web proc types

### ISSUE THREAD

**Issue by michaelshobbs** [dokku/dokku#2125]
Should we modify the deploy flow such that for non-web proc types we don't run a second container to provide "zero-downtime?"

This question came up in a couple of my production environments and I know I've seen this on IRC once or twice. I'm thinking it makes sense to stop non-web proc containers before starting the new ones.

Thoughts here?

**Comment by josegonzalez** [comment]
Possibly related to #2073. We could do something like:

```shell
dokku checks:disable <app> <process-type>
```

Thoughts?

**Comment by michaelshobbs** [comment]
Does that mean we're ok with #2073 in that `checks:disable` will actually disable overlapping containers? Also how would you proposed structure this in such a way that we can disable checks for a specific proc type?

Of note: I opened a new RFC specifically because I wanted to discuss the option of **only** running overlapping containers for **web processes**.

**Comment by josegonzalez** [comment]
Yeah I think that's the point of `checks:disable`, no? I mean honestly it's a bad name for the thing - `zero-downtime:disable` is only marginally better - but I thought that was the point of the command change in the RFC.

Regarding the command, the `process-type` arg would be optional, otherwise it's every process. We could be cute and do comma-separated process types.

**Comment by michaelshobbs** [comment]
Agreed on the naming bit. 

Regarding `process-type`, my question was more around how we store which process types vs no process types. Thoughts there?

**Comment by josegonzalez** [comment]
`*` for everything, otherwise specify a specific process type?

**Comment by michaelshobbs** [comment]
`*` seems like it might be sketchy to test for in bash land, no? Maybe `all`? I'd be surprised if someone named a proc `all`.

**Comment by josegonzalez** [comment]
Fine by me.

**Comment by michaelshobbs** [comment]
Closing in favor of #2073
